src/librustc/traits/select.rs

```diff
@@ -534,6 +540,7 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
         assert!(!obligation.predicate.has_escaping_regions());
 
         let stack = self.push_stack(TraitObligationStackList::empty(), obligation);
+        // FIXME: catch and report overflow error here as well?
```

Comment:
I'd prefer to see Overflow propagated out here as a separate sort of error, yes; not sure if that will make a bunch of other stuff awkward?

Reply:
So just to be clear, currently it's in there as a possible `SelectionError::Overflow`, and the equivalent of current behavior would be to catch that particular variant and report it right away (which this code is currently not doing). But by separate sort of error I suppose you mean change the return type from `_` to `Result<_, OverflowError>`, and then have the callers of `select` handle it? If so I think that'd definitely make a bunch of other stuff awkward.

I personally think what makes most sense is to catch the overflow variant and report it right here, because that would keep the overflow-handling business local to `select`, as it has been so far.